src/main/java/io/quarkus/security/identity/RunAsUser.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.security.identity;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.security.Principal;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Identifies a method that should be executed with a new, temporary {@link SecurityIdentity}.
+ * This annotation may only be applied in supported scenarios specified in Quarkus documentation.
```

Comment:
```suggestion
```

Referring here to some undoc-ed scenarios probably does not clarify anything, this is just an API, we don't make any promises here that for ex it would apply to all security cases